torchdynamo/variables/tensor.py

```diff
@@ -269,6 +269,20 @@ def context():
                     need_unwrap=False,
                     **options,
                 )
+        elif isinstance(example_value, contextlib._GeneratorContextManager):
```

Comment:
I don't think we can safely do this for all context managers.  We could have a list of context managers where we have manually confirmed it is safe for though.

Reply:
Sg, I can narrow it down to a subset.

Reply:
rm